### ISSUE TITLE
Tangens computation not finished on complex argument

### ISSUE THREAD

**Issue by w31t1** [aleaxit/gmpy#298]
## Short description
The `tan` computation does not finished on distinct complex arguments. Thread has to be stopped manually.

## Steps for reproduce:
```
import numpy as np
import gmpy2
zi = np.float64(54542684160.0)
zr = np.float64(1.401298464324817e-45)
gmpy2.mpc(np.float64(zr), np.float64(zi))
Out[7]: mpc('1.4012984643248171e-45+54542684160.0j')
gmpy2.tan(gmpy2.mpc(np.float64(zr), np.float64(zi)))
^C
```

## Expected result
```
mpc('0.0+1.0j')
```
(Result on version `2.0.8` )

Note: Wolfram Alpha returns [complex infinity](https://www.wolframalpha.com/input/?i=ctan%28%2B1.401298464324817e-45%2B5.4542684160e%2B10i%29), MPC 1.2 returns `(NaN NaN)`

## Affected versions
Tested on beta versions `2.1.0b5` and `2.1.0b6` and current `master` (e1c258c2dcaab88453d2627bd40f00875b511637).

**Comment by casevh** [comment]
I understand the issue. I'm working on a fix.

**Comment by casevh** [comment]
Fix to restore 2.0.8 behavior has been committed. Thanks for the report.
